ethereum-bridge-contracts/contracts/InboundChannel.sol

```diff
@@ -84,7 +106,18 @@ contract InboundChannel is AccessControl, ISimplifiedMMRProof, ReentrancyGuard {
             "insufficient gas for delivery of all messages"
         );
 
-        processMessages(payable(msg.sender), batch.messages);
+        uint256 results = processMessages(payable(msg.sender), batch.messages);
+
+        uint256 gas_used = block.gaslimit - gasleft();
+        gas_proofs[batch_nonce] = keccak256(abi.encodePacked(gas_used, block.basefee));
```

Comment:
You should add the same data here as to BatchDispatched event. Also `encodePacked` could open vulnerability, so I think it's better to use `abi.encode`